viewing_party/party.py

```diff
@@ -1,23 +1,152 @@
 # ------------- WAVE 1 --------------------
 
+from re import U
+from typing import Counter
+
+
 def create_movie(title, genre, rating):
-    pass
+    if not all([title, genre, rating]):
+        return
+
+    return {
+        "title": title,
+        "genre": genre,
+        "rating": rating
+    }
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+    
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(data, movie_title):
+    item = None
+    for movie in data['watchlist']:
+        if movie['title'] == movie_title:
+            item = movie
+            break
+    if item is None:
+        return data
+
+    data['watchlist'].remove(item)
+    data['watched'].append(item)
+
+    return data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(data):
+    total_rating = 0
+    for movie in data["watched"]:
+        total_rating += movie["rating"]
+    if len(data["watched"]) == 0:
+        return 0
+    return total_rating /len(data["watched"])
+
+def get_most_watched_genre(data):
+    
+    counter = {}
+    for movie in data["watched"]:
+        if movie["genre"] not in counter:
+            counter[movie["genre"]] = 0
+        counter[movie["genre"]] += 1
+
+    max_genre = None
+    max_freq = 0
+    for genre, frequency in counter.items():
+        if frequency >= max_freq:
+            max_genre = genre
+            max_freq = frequency
+
+    return max_genre
 
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_unique_watched(data):
+    unique = []
+    friends_list_watched_movies = []
+
+    for friend in data["friends"]:
+        for movie in friend["watched"]:
+            friends_list_watched_movies.append(movie['title'])
 
+    for movie in data["watched"]:
+        if movie["title"] not in friends_list_watched_movies:
+            unique.append(movie)
+
+    return unique
+
+def get_friends_unique_watched(data):
+    title_list = []
+    for movie in data["watched"]:
+        title_list.append(movie["title"])
+    
+    unique_watched_titles = set()
+    unique_watched = []
+    for friend in data["friends"]:
+        for movie in friend["watched"]:
+            if movie["title"] not in title_list and movie["title"] not in unique_watched_titles:
+                unique_watched.append(movie)
+                unique_watched_titles.add(movie["title"])
+                
+                
+
+    return unique_watched

```

Comment:
This implementation works and also accomplishes the requirements for get_friends_unique_watched! Nice work!

 I have a suggestion to help simplify the logic a little bit. 

I see that you're using unique_watched_titles in line 96 to check that you are not adding movies more than once to your unique_watched list. Since you are only returning unique_watched, you can write this logic without using unique_watched_titles. 

If you don't use unique_watched_titles, you can remove lines 92 and line 98 and wouldn't need title_list too since you don't need the title strings to compare. 

A refactored version could look like:
```python 
unique_watched = []
    for friend in data["friends"]:
        for movie in friend["watched"]:
            if movie not in data["watched"] and movie not in unique_watched:
                unique_watched.append(movie)
            
    return unique_watched
```

I'm happy to talk about this refactor if you have any questions about it!